### ISSUE TITLE
3D Shape objects lags when set position on a imagepoint

### ISSUE THREAD

**Issue by uDevah** [Scirra/Construct-bugs#7508]
## Problem description

If you set position with another 3D Shape based on a imagepoint, this objects lags.

## Attach a .c3p

[position_bug_helper.zip](https://github.com/Scirra/Construct-bugs/files/13205357/position_bug_helper.zip)

## Steps to reproduce

1. Open the project
2. Move in any direction with arrow keys

## Observed result

There is a text in the top-left corner that show the distance between the 3D_Shape and a sprite object.
On move with arrow keys the distance changes, meaning the sprite position based on a 3D Shape image point lags.

## Expected result

The distance between the two objects should be always the same.

## More details

In the project there are events to easily test the sprite position to 3D Shape with just coordinates and there is 0 lag.
Also an sprite that sets position with another via imagepoints and 0 lag.
But when you set an object to 3D Shape position based on any image points it lags.
The lag also happens with FaceImagePoints.

-

**Affected browsers/platforms:** 
Tested on Chrome. Windows 10.

**Comment by AshleyScirra** [comment]
This project uses a third-party addon:

> Effect Layer Light 3D (Mikal_LayerLight3D) by Mikal

Please provide a project not using any third-party addons as per the bug report guidelines.

**Comment by AshleyScirra** [comment]
This is not a bug: using 'Set position' when the movement is controlled by a behavior can often end up with a lag, as whichever one steps first will be one frame ahead. The solution is to use the hierarchies feature which was specifically designed to solve this problem. For example use 'On start of layout' to set the "hand" position and then add it as a child of the Hero_3D object, and it works with no lag.

**Comment by uDevah** [comment]
I tried moving the object "Hero_Col" with just set position to "self.X + 15" pressing a key, without any behaviours, and it lags too with imagepoints and faceimagepoints but not with any other methods. Tried with other behaviours like car, custom, platforms and the same.

Set position with imagepoints/faceimagepoints works in all cases i tried without lag except if a 3D Shape, at least in this particular case.

But yes, adding as a child fixes the problem, thanks.
